### PR TITLE
[4.3] HELP-14771: Add stepswitch.min_shortdial_destination parameter

### PR DESCRIPTION
**Pull request by harenson**
And also refactor short-dial correction code in order to be able to add
unit tests for that functionality per @jamesaimonetti request.

Master PRs:
- https://github.com/2600hz/kazoo-stepswitch/pull/4
- https://github.com/2600hz/kazoo-crossbar/pull/21

### CODE REVIEW

File: applications/stepswitch/src/stepswitch_util.erl [hunk 1]
```diff
@@ -91,41 +101,58 @@ correct_shortdial(<<"+", Number/binary>>, CIDNum, DeniedCallRestrictions) ->
...
+               ,[Number, byte_size(Number), Length]
+               ),
+    'undefined'.
+
+-spec should_correct_shortdial(kz_term:ne_binary()) -> {boolean(), pos_integer()}.
+should_correct_shortdial(Number) ->
+    Length = kapps_config:get_integer(?SS_CONFIG_CAT, <<"shortdial_correction_if_length_is">>, 0),
+    %% The length is also included on the response just to avoid having to read this configuration
+    %% parameter again when printing the debug line. So it is only useful for the `{false, N}' branch.
+    {0 =:= Length
```
Comment (jamesaimonetti) [inline applications/stepswitch/src/stepswitch_util.erl:131]: I would prefer the logic portion (comparing Length and Number) in its own function:

```erlang
should_correct_shortdial(Number) ->
    Length = kapps_config:(...),
    should_correct_shortdial(Number, Length).

should_correct_shortdial(_Number, 0) -> {'true', 0};
should_correct_shortdial(Number, Length) ->
    {byte_size(Number) =:= Length, Length}.
```

This is also easily testable without using meck or other means.

File: applications/crossbar/priv/api/swagger.json [hunk 2]
```diff
@@ -35262,16 +35262,21 @@
...
                 "min_shortdial_correction": {
                     "default": 2,
-                    "description": "stepswitch minimum shortdial correction",
+                    "description": "add at least {MIN} digits from caller ID to destination number",
                     "minimum": 1,
                     "type": "integer"
                 },
+                "min_shortdial_destination": {
+                    "default": 0,
+                    "description": "do short-dial correction if dialed number's length is equal to this value ONLY when it is greater than 0",
```
Comment (jamesaimonetti) [inline applications/crossbar/priv/api/swagger.json:35277]: I still think this should not default to 0. Instead, I think you could put a `minumum:1` constraint. So if `min_shortdial_destination` is unset/null, it is ignored. If set, the value must be greater than 0 already so use it.

Comment (harenson) [reply applications/crossbar/priv/api/swagger.json:35277]: Yeah, I forgot to remove that, the code is not relying on `0` as the default value anymore, that's a leftover from the previous changes.

File: applications/stepswitch/src/stepswitch_util.erl [hunk 3]
```diff
@@ -91,61 +101,100 @@ correct_shortdial(<<"+", Number/binary>>, CIDNum, DeniedCallRestrictions) ->
...
-            'undefined';
-        'false' ->
-            CorrectedNumber
-    end.
+    lager:debug("corrected shortdial ~s via CID ~s to ~s", [Number, CIDNum, CorrectedNumber]),
+    CorrectedNumber.
 
--spec should_deny_reclassified_number(kz_term:ne_binary(), kz_json:object()) -> boolean().
-should_deny_reclassified_number(CorrectedNumber, DeniedCallRestrictions) ->
+-spec maybe_deny_reclassified_number(kz_term:ne_binary(), kz_term:ne_binary(), kz_json:objects()) -> kz_term:api_binary().
```
Comment (jamesaimonetti) [inline applications/stepswitch/src/stepswitch_util.erl:149]: Spec is wrong, `DeniedCallRestrictions` is a `kz_json:object()` no? At least, previous calls have that var tagged as such.

Comment (jamesaimonetti) [reply applications/stepswitch/src/stepswitch_util.erl:149]: Return should be `kz_term:api_ne_binary()` as well.

**Pull request merged**